plasmapy/simulation/particle_tracker.py

```diff
@@ -219,37 +290,81 @@ def save_now(self) -> bool:
         """Determine if to save on the current push step."""
         ...
 
+    @property
+    def results(self) -> dict[str, u.Quantity]:
+        """Return the results of the simulation.
+        The quantities returned are the times, positions, and velocities, respectively.
+        """
+        return self._apply_units_to_results()
+
     def save(self) -> None:
-        """Save the current state of the simulation to disk or memory based on whether the output directory was set."""
+        # """Save the current state of the simulation to disk or memory based on whether the output directory was set."""
+        """Save the current state of the simulation to memory."""
+
+        self._save_to_memory()
 
         if self.output_directory is not None:
             self._save_to_disk()
-        else:
-            self._save_to_memory()
 
     def _save_to_disk(self) -> None:
         """Save a hdf5 file containing simulation positions and velocities."""
+        #
+        # path = self.output_directory / f"{self.tracker.iteration_number}.hdf5"
+        #
+        # with h5py.File(path, "w") as output_file:
+        #     output_file["x"] = self.tracker.x
+        #     output_file["v"] = self.tracker.v
 
         path = self.output_directory / f"{self.tracker.iteration_number}.hdf5"
+        attributes = self._results.get("attributes", [])
 
         with h5py.File(path, "w") as output_file:
-            output_file["x"] = self.tracker.x
-            output_file["v"] = self.tracker.v
-
+            for key, value in self._results.items():
+                if key == "attributes":
+                    continue
+
+                # Should this entry be saved as an attribute (magnification, number of particles)
+                # or a dataset (positions, velocities)
+                if key in attributes:
+                    output_file.attrs.create(key, value)
+                else:
+                    output_file.create_dataset(key, data=value)
+
+    # TODO: Find a better name for this method
     def _save_to_memory(self) -> None:
-        """Append simulation positions and velocities to save routine object."""
-        self.x_all.append(np.copy(self._particle_tracker.x))
-        self.v_all.append(np.copy(self._particle_tracker.v))
+        """Update the results dictionary with the current state of the simulation."""
+        # """Append simulation positions and velocities to save routine object."""
+        # self.x_all.append(np.copy(self._particle_tracker.x))
+        # self.v_all.append(np.copy(self._particle_tracker.v))
+
+        self._results["x"].append(np.copy(self._particle_tracker.x))
```

Comment:
Results is understandably a dict of lists, since we are appending to them as the simulation runs. But, when the user accesses this property, I think it would be more convenient for plotting etc. to return a dict of Quantities? I think your unit application already does this? 

Interesting: I didn't know that list * quantity = quantity without having to do the conversion list -> np.ndarray first.

Reply:
Yep this is what I ahve:
```py

    @property
    def results(self) -> dict[str, u.Quantity]:
        """Return the results of the simulation.
        The quantities returned are the times, positions, and velocities, respectively.
        """
        return self._apply_units_to_results()
   ```

Is that the behavior you were looking for?

Definitely a plus that astropy handles the list to array conversion!